src/Console/ConsoleIo.php

```diff
@@ -250,6 +250,21 @@ public function success($message = null, $newlines = 1, $level = self::NORMAL)
         return $this->out($message, $newlines, $level);
     }
 
+    /**
+     * Halts the the current process with a StopException.
+     *
+     * @param string $message Error message.
+     * @param int $code Error code.
+     * @return void
+     * @throws \Cake\Console\Exception\StopException
+     */
+    public function abort($message, $code = Command::CODE_ERROR)
+    {
+        $this->error($message);
+
+        throw new StopException($code);
```

Comment:
The 1st argument of `StopException`'s constructor is string `$message` not int `$code`. This creates type error doing 3.x => 4.x merge.

Reply:
This is why I prefer all future enhancements to be done on 4.x and backported to 3.next. It avoids such errors and problems during 3.x => 4.x merges.

Reply:
I changed it to `throw new StopException($message, $code);` in 4.x.

Reply:
We have a much deeper underlying issue here IMO we should fix instead.
The first argument, message is not used at all right now which is really bad. It forces one to always add a ->error() before, which makes no sense since the message argument is there.
=> https://github.com/cakephp/cakephp/pull/13588